### PR TITLE
Crude shotgun

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

SUMMARY: Content "Add new shotgun to replace the obsolete pipe shotgun"

#### Purpose of change

I wanted a low-level shotgun to replace the obsoleted pipe weapons: this is one example. Simple enough to make with basic tools, but awfully shoddy. Not much to say besides that.

#### Describe the solution

This weapon's recipe is unlocked at level 1 marksmanship & level 1-2 mechanics. Also needs a welding tool for the end cap.

#### Describe alternatives you've considered

Not doing this.

#### Testing

Checked the JSON against other stuff.
Tried wishing for it, reloaded etc.

#### Additional context

Side note: is the 2 Shot Special still in the game?

### CODE REVIEW

File: data/json/items/gun/shot.json [hunk 1]
```diff
@@ -766,4 +766,32 @@
...
+    "clip_size": 1,
+    "modes": [ [ "DEFAULT", "single", 1 ] ],
+    "valid_mod_locations": [
+      [ "barrel", 1 ],
+      [ "sling", 1 ],
+      [ "brass catcher", 1 ],
+      [ "sights mount", 1 ],
+      [ "underbarrel", 1 ]
+    ],
+    "flags": [ "RELOAD_ONE" ],
```
Comment (LaVeyanFiend) [inline data/json/items/gun/shot.json:794]:
```suggestion
      [ "sights mount", 1 ],
      [ "underbarrel mount", 1 ]
    ],
    "flags": [ "RELOAD_EJECT" ],
```
1. Brass catcher only applies to guns with extractors. Slamfire shotgun gets RELOAD_EJECT.
2. No native Picatinny or NATO accessory rail on the bottom = it needs a mount slot.

**Comment by Xenomorph-III** [comment]
Please do tell me if I messed up: this was a spur-of-the moment addition.

File: data/json/items/gun/shot.json [hunk 2]
```diff
@@ -766,4 +766,32 @@
...
   }
+  {
+    "id": "makeshift_shotgun",
+    "copy-from": "shotgun_base",
+    "type": "GUN",
+    "name": { "str": "makeshift shotgun" },
+    "description": "A crude shotgun, composed of two thick steel pipes, an end cap and a nail. The lack of sights make this weapon only useful at point-blank range.",
+    "weight": "3629 g",
+    "volume": "2264 ml",
+    "looks_like": "remington_870",
```
Comment (LaVeyanFiend) [inline data/json/items/gun/shot.json:777]:
```suggestion
    "looks_like": "pipe",
```
If it doesn't have a stock, it's going to look like a pipe.

File: data/json/items/gun/shot.json [hunk 3]
```diff
@@ -766,4 +766,32 @@
...
     "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "shot": 6 } } ]
   }
+  {
+    "id": "makeshift_shotgun",
+    "copy-from": "shotgun_base",
+    "type": "GUN",
+    "name": { "str": "makeshift shotgun" },
+    "description": "A crude shotgun, composed of two thick steel pipes, an end cap and a nail. The lack of sights make this weapon only useful at point-blank range.",
+    "weight": "3629 g",
+    "volume": "2264 ml",
```
Comment (LaVeyanFiend) [inline data/json/items/gun/shot.json:776]: Seems like there should be some compression here, given one fits at least partially into the other.

File: data/json/recipes/recipe_weapon.json [hunk 4]
```diff
@@ -2784,4 +2784,28 @@
...
+      [
+        [ "oxy_torch", 6 ],
+        [ "welder", 30 ],
+        [ "welder_crude", 45 ],
+        [ "toolset", 45 ],
+        [ "small_repairkit", 150 ],
+        [ "large_repairkit", 30 ]
+      ]
+    ],
+    "components": [ [ [ "pipe", 2 ] ], [ [ "nail", 1 ] ], [ [ "scrap", 2 ] ] ]
```
Comment (LaVeyanFiend) [inline data/json/recipes/recipe_weapon.json:2809]: I'd personally add one more piece of scrap to cover the sling swivel, you can ignore this suggestion if you want.

Comment (Xenomorph-III) [reply data/json/recipes/recipe_weapon.json:2809]: What's a "sling swivel"? just curious.

Comment (LaVeyanFiend) [reply data/json/recipes/recipe_weapon.json]: image
You feed a sling through the loop.

File: data/json/items/gun/shot.json (hunk 3 above)
Comment (Xenomorph-III) [reply data/json/items/gun/shot.json:776]: Any suggestions?

"A crude shotgun composed of two thick steel pipes, an end cap and a nail. It is only useful at point blank range." it my slightly shortened version.

File: data/json/recipes/recipe_weapon.json (hunk 4 above)
Comment (Xenomorph-III) [reply data/json/recipes/recipe_weapon.json]: Got it, thanks. It's added.

**Comment by dissociativity** [comment]
could copper tubing work feaseably? that's not too hard to alter the size of

**Comment by Xenomorph-III** [comment]
The copper tubing description explicitly states it's thin, so it'll likely explode.

**Comment by Xenomorph-III** [comment]
Trying to lint gives this:

Capture

File: data/json/recipes/recipe_weapon.json [hunk 5]
```diff
@@ -2784,4 +2784,28 @@
     "qualities": [ { "id": "CUT", "level": 1 }, { "id": "HAMMER", "level": 1 } ],
     "components": [ [ [ "leather", 30 ], [ "rope_superior_short", 2, "LIST" ] ], [ [ "scrap", 9 ], [ "razor_blade", 9 ] ] ]
   }
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_weapon.json:2786]: Maybe the linter's not taking a syntax error too well?
```suggestion
  },
```

File: data/json/items/gun/shot.json [hunk 6]
```diff
@@ -766,4 +766,31 @@
     "flags": [ "RELOAD_ONE" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "shot": 6 } } ]
   }
```
Comment (anothersimulacrum) [inline data/json/items/gun/shot.json:768]:
```suggestion
  },
```

File: data/json/recipes/recipe_weapon.json [hunk 7]
```diff
@@ -2784,4 +2784,28 @@
...
+        [ "oxy_torch", 6 ],
+        [ "welder", 30 ],
+        [ "welder_crude", 45 ],
+        [ "toolset", 45 ],
+        [ "small_repairkit", 150 ],
+        [ "large_repairkit", 30 ]
+      ]
+    ],
+    "components": [ [ [ "pipe", 2 ] ], [ [ "nail", 1 ] ], [ [ "scrap", 3 ] ] ]
+  }3
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_weapon.json:2810]:
```suggestion
  }
```

**Comment by Xenomorph-III** [comment]
I was thinking of making this Mechanics 1 instead of 2: it's even simpler than the old pipe shotgun.

**Comment by Xenomorph-III** [comment]
Again with the request too long error... this time on shot.json.

**Comment by Xenomorph-III** [comment]
Why are all the checks still failing?

**Comment by Xenomorph-III** [comment]
I've already re-opened another pull request for this: this one will be closed for now.

**Pull request closed without merging**